fdbserver/fdbserver.actor.cpp

```diff
@@ -1786,6 +1796,8 @@ int main(int argc, char* argv[]) {
 			if (!dataFolder.size())
 				dataFolder = format("fdb/%d/", publicAddresses.address.port);  // SOMEDAY: Better default
 
+			auto histogramReportActor = histogramReport();
```

Comment:
histogramReportActor needs to be wait()'ed on.  You can `push_back` it into `actors` (two lines below this), since that's passed into `stopAfter( waitForAll())`.

(Similar logic exists next to the other call to histogramReport().

Reply:
Actually `wait` is not always needed, @jzhou77 mentioned this to me earlier in slack talk. A quick test:

```c++
#include <iostream>
#include <vector>

#include "flow/flow.h"
#include "flow/Platform.h"
#include "flow/TLSConfig.actor.h"
#include "flow/actorcompiler.h"

ACTOR Future<Void> func() {
	state int v = 0;
	loop {
		wait(delay(0.5));

		std::cout << " ---- " << v++ << std::endl;
	}
}


ACTOR Future<Void> testFunc() {
	state int v = 0;
	loop {
		if (++v == 10) break;

		wait(delay(1.0));

		std::cout << " Hello World " << v << std::endl;
	}

	return Void();
}


int main() {
	platformInit();
	g_network = newNet2(TLSConfig(), false, true);

	auto __ = func();
	auto _ = stopAfter(waitForAll<Void>({testFunc()}));
	g_network->run();

	return 0;
}
```

And the output is

```bash
<@foundationdb-tutorial/build>-<⎇ tutorial>-<±>-<*>-> bin/extra_tutorials/async_state
 ---- 0
 Hello World 1
 ---- 1
 ---- 2
 Hello World 2
 ---- 3
 ---- 4
 Hello World 3
 ---- 5
 ---- 6
 Hello World 4
 ---- 7
 ---- 8
 Hello World 5
 ---- 9
 ---- 10
 Hello World 6
 ---- 11
 ---- 12
 Hello World 7
 ---- 13
 ---- 14
 Hello World 8
 ---- 15
 ---- 16
 Hello World 9
<@foundationdb-tutorial/build>-<⎇ tutorial>-<±>-<*>->
```

Reply:
I think `push_back` to `actors` is the preferred way so that if the actor `histogramReport` throws an error, the error can be caught.

Reply:
Everything in actors ends up being waited on.  I think we agree.

Reply:
It's good to know that it spawns before wait is invoked though.  Thanks.

Reply:
I checked the code, for the simulator part there is not a vector of actors, nor a `stopAfter(waitForAll())` call, so I am leaving it there.